include/boost/json/detail/impl/object_impl.ipp

```diff
@@ -70,36 +70,31 @@ void
 object_impl::
 build() noexcept
 {
-    auto end = this->end();
-    for(auto p = begin(); p != end;)
+    if(! size())
+        return;
+    const auto first = begin() - 1;
+    auto last = end() - 1;
+    for(; last > first; --last)
```

Comment:
Can we avoid the extra conditional with this
```
auto const first = begin();
auto last = end();
for(; last-- > first;)
{
    ...
```
?

Reply:
No, we still need to do an iteration when `last == first`

Reply:
Okay that was no good, this should work though:
```
7
// must work when table pointer is null
auto const first = begin();
for( auto last = end(); last > first; )
{
    --last;
    ...
```

Reply:
Did you change this on your branch or should I change it here?

Reply:
I prefer not to make these kinds of changes on my own, because then it can be confusing if you are also making the change, but I already merged it, and maybe you found a bug so now you have a branch that is different than mine. Since you're the lead on this pull request, it is better if you are the only one applying changes to it.

Reply:
Copy that